### ISSUE TITLE
Each parameter of `nn.MaxPool1d()`, `nn.MaxPool2d()` and `nn.MaxPool3d()` should have `required` or `optional`

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#136096]
### 📚 The doc issue

The docs of [nn.MaxPool1d()](https://pytorch.org/docs/stable/generated/torch.nn.MaxPool1d.html), [nn.MaxPool2d()](https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html) and [nn.MaxPool3d()](https://pytorch.org/docs/stable/generated/torch.nn.MaxPool3d.html) don't say which parameters are required or optional as shown below:

> Parameters
> - kernel_size ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]]) – The size of the sliding window, must be > 0.
> - stride ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]]) – The stride of the sliding window, must be > 0. Default value is kernel_size.
> - padding ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]]) – Implicit negative infinity padding to be added on both sides, must be >= 0 and <= kernel_size / 2.
> - dilation ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]]) – The stride between elements within a sliding window, must be > 0.
> - return_indices ([bool](https://docs.python.org/3/library/functions.html#bool)) – if True, will return the max indices along with the outputs. Useful for [torch.nn.MaxUnpool2d](https://pytorch.org/docs/stable/generated/torch.nn.MaxUnpool2d.html#torch.nn.MaxUnpool2d) later
> - ceil_mode ([bool](https://docs.python.org/3/library/functions.html#bool)) – when True, will use ceil instead of floor to compute the output shape

### Suggest a potential alternative/fix

So, each parameter should have `required` or `optional` something like as shown below:

> Parameters
> - kernel_size ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]], required) – The size of the sliding window, must be > 0.
> - stride ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]], optional) – The stride of the sliding window, must be > 0. Default value is kernel_size.
> - padding ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]], optional) – Implicit negative infinity padding to be added on both sides, must be >= 0 and <= kernel_size / 2.
> - dilation ([Union](https://docs.python.org/3/library/typing.html#typing.Union)[[int](https://docs.python.org/3/library/functions.html#int), [Tuple](https://docs.python.org/3/library/typing.html#typing.Tuple)[[int](https://docs.python.org/3/library/functions.html#int)]], optional) – The stride between elements within a sliding window, must be > 0.
> - return_indices ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – if True, will return the max indices along with the outputs. Useful for [torch.nn.MaxUnpool2d](https://pytorch.org/docs/stable/generated/torch.nn.MaxUnpool2d.html#torch.nn.MaxUnpool2d) later
> - ceil_mode ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – when True, will use ceil instead of floor to compute the output shape
